proxy/src/console/provider/neon.rs

```diff
@@ -278,9 +278,15 @@ impl super::Api for Api {
         // The connection info remains the same during that period of time,
         // which means that we might cache it to reduce the load and latency.
         if let Some(cached) = self.caches.node_info.get(&key) {
-            info!(key = &*key, "found cached compute node info");
-            ctx.set_project(cached.aux.clone());
-            return Ok(cached);
+            let (cached, info) = cached.take_value();
```

Comment:
This block of code is essentially the same as lines 297ff. Put that into a function?

Reply:
I much prefer WET (write everything twice) over DRY (don't repeat yourself). Nuance of course but I don't think there's much benefit with writing a dedicated function just for this